### PR TITLE
Siamrpn train

### PR DESCRIPTION
**Pull request by FrankYoungchen**
Siamrpn train and  fix some bug before

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1220-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1220/2/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1220-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1220/3/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/data/tracking_data/track.py [hunk 1]
```diff
@@ -252,18 +252,18 @@ class TrkDataset(dataset.Dataset):
...
+                               'det/train.json'),
+                 dataset_frame_range=(100, 3, 1, 1),
                  dataset_num_use=(100000, -1, -1, -1),
                  train_search_size=255,
                  train_exemplar_size=127,
                  anchor_stride=8,
                  anchor_ratios=(0.33, 0.5, 1, 2, 3),
-                 train_base_size=8,
-                 train_output_size=25,
+                 train_base_size=0,
```
Comment (bryanyzhu) [inline gluoncv/data/tracking_data/track.py]: What is train_base_size, and why it is 0?

Comment (FrankYoungchen) [reply gluoncv/data/tracking_data/track.py]: in pyost SiamRPN, train_base_size will help get output feature map size.Formula is 
```
score_size = (cfg.TRACK.INSTANCE_SIZE - cfg.TRACK.EXEMPLAR_SIZE) // \
            cfg.ANCHOR.STRIDE + 1 + cfg.TRACK.BASE_SIZE
```

**Comment by mli** [comment]
Job PR-1220-6 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1220/6/index.html 
Code coverage of this PR:  vs. Master:

**Comment by bryanyzhu** [comment]
@zhreshold @hetong007 This PR is to add model training for single object tracking. We have completed tutorial, train&test, unit test, etc., we will update the model zoo soon. Please help review the PR if you are interested. 

One minor thing is, our training is slower compared to PyTorch counterpart, but let me take care of this in the future.

File: gluoncv/model_zoo/siamrpn/siam_rpn.py [hunk 2]
```diff
@@ -26,13 +27,19 @@ class DepthwiseXCorr(HybridBlock):
...
+        if_train : str
+            if_train is True if train, False if test.
         kernel_size : float
-            hidden kernel size
+            hidden kernel size.
+        ctx : mxnet.Context
+            Context such as mx.cpu(), mx.gpu(0).
     """
-    def __init__(self, hidden, out_channels, kernel_size=3):
+    def __init__(self, hidden, out_channels, bz=1, if_train=False, kernel_size=3, ctx=cpu()):
```
Comment (zhreshold) [inline gluoncv/model_zoo/siamrpn/siam_rpn.py]: nit: `if_train` -> `is_train`

File: gluoncv/utils/filesystem.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ def try_import_cv2():
     return try_import('cv2', msg)
 
 def try_import_colorama():
-    """Try import cv2 at runtime.
+    """Try import colorama at runtime.
```
Comment (zhreshold) [inline gluoncv/utils/filesystem.py]: why is colorma required?

File: scripts/tracking/train.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+
+from mxnet import gluon, nd, context, autograd
+from mxnet.contrib import amp
+
+from gluoncv.utils import LRScheduler, LRSequential, split_and_load
+from gluoncv.data.tracking_data.track import TrkDataset
+from gluoncv.model_zoo import get_model
+from gluoncv.loss import SiamRPNLoss
+
+os.system("export MXNET_CUDNN_AUTOTUNE_DEFAULT=0")
```
Comment (zhreshold) [inline scripts/tracking/train.py]: I think this line should be moved to line 13 before mxnet import

File: gluoncv/utils/filesystem.py (hunk 3 above)
Comment (FrankYoungchen) [reply gluoncv/utils/filesystem.py]: in benchmark,I will add import colorma

File: gluoncv/model_zoo/siamrpn/siam_rpn.py (hunk 2 above)
Comment (FrankYoungchen) [reply gluoncv/model_zoo/siamrpn/siam_rpn.py]: ok

File: scripts/tracking/train.py (hunk 4 above)
Comment (FrankYoungchen) [reply scripts/tracking/train.py]: ok

**Comment by mli** [comment]
Job PR-1220-10 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1220/10/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/data/tracking_data/track.py [hunk 5]
```diff
@@ -546,10 +546,12 @@ def select(position, keep_num=16):
         delta[2] = np.log(tw / w)
         delta[3] = np.log(th / h)
 
-        overlap = Iou([x1, y1, x2, y2], target)
+        target_temple = np.array([target[0], target[1], target[2], target[3]]).reshape(1, -1)
```
Comment (hetong007) [inline gluoncv/data/tracking_data/track.py]: What does "temple" mean here?

File: scripts/tracking/test.py [hunk 6]
```diff
@@ -28,12 +28,10 @@ def parse_args():
                         help='whether visualzie result')
     parser.add_argument('--mode', type=str, default='hybrid',
                         help='mode in which to train the model.options are symbolic, hybrid')
-    parser.add_argument('--num_gpus', type=int, default=0,
+    parser.add_argument('--num-gpus', type=int, default=0,
```
Comment (hetong007) [inline scripts/tracking/test.py]: please be consistent of whether using `-` or `_`, for all parser arguments.

**Comment by mli** [comment]
Job PR-1220-11 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1220/11/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**